sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -529,18 +530,32 @@ def set_read_only(
             404: ResourceNotFoundError
         }
 
+        match_condition = kwargs.pop("match_condition", MatchConditions.Unconditionally)
+        if match_condition == MatchConditions.IfNotModified:
+            error_map[412] = ResourceModifiedError
+        if match_condition == MatchConditions.IfModified:
+            error_map[412] = ResourceNotModifiedError
+        if match_condition == MatchConditions.IfPresent:
+            error_map[412] = ResourceNotFoundError
+        if match_condition == MatchConditions.IfMissing:
+            error_map[412] = ResourceExistsError
+
         try:
             if read_only:
                 key_value = self._impl.put_lock(
                     key=configuration_setting.key,
                     label=configuration_setting.label,
+                    if_match=prep_if_match(configuration_setting.etag, match_condition),
```

Comment:
How about checking if users have explicitly provided "if_match" kwarg? It would avoid breaking existing users who pass if_match argument.

Reply:
We don't officially support "if_match" in our SDK. Supporting both MatchCondition & "If_match" will confuse customers.

Reply:
Makes sense, thanks!